src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm_copy_b.cpp

```diff
@@ -0,0 +1,316 @@
+#endif
+}
+
+void BrgemmCopyBKernel::execute(matmul::jit_brgemm_matmul_copy_b_t* kernel, const void* src, const void* dst, const void* comp, size_t N, size_t K) {
+    auto ctx = matmul::jit_brgemm_matmul_copy_b_t::ctx_t();
+    ctx.current_N_blk = N;
+    ctx.src = src;
+    ctx.tr_src = dst;
+    ctx.compensation_ptr = comp;
+    ctx.zp_a_compensation_ptr = nullptr;
+    ctx.zp_a_neg_value_ptr = nullptr;
+    ctx.current_K_start = 0;
+    ctx.current_K_iters = K;
+
+    OV_CPU_JIT_EMITTER_ASSERT(kernel, "Kernel hasn't been created");
+    (*kernel)(&ctx);
+}
+
+BrgemmCopyBKernelExecutor::BrgemmCopyBKernelExecutor(ov::intel_cpu::MultiCacheWeakPtr kernel_cache, BrgemmCopyBKernelConfig config)
+    : CPUKernelExecutor<BrgemmCopyBKernelConfig, BrgemmCopyBKernel>(std::move(kernel_cache), std::move(config)) { }
+
+std::shared_ptr<BrgemmCopyBKernel> BrgemmCopyBKernelExecutor::compile_kernel(const BrgemmCopyBKernelConfig& config) const {
+    std::shared_ptr<BrgemmCopyBKernel> compiled_kernel = std::make_shared<BrgemmCopyBKernel>();
+    // BrgemmCopyB is not executable - nothing to compile
+    if (!config.is_empty()) {
+        compiled_kernel = std::make_shared<BrgemmCopyBKernel>(config);
+        OV_CPU_JIT_EMITTER_ASSERT(compiled_kernel, "compiled kernel is nullptr");
+        compiled_kernel->create_kernel();
+    }
+
+    return compiled_kernel;
+}
+
+void BrgemmCopyBKernelExecutor::update_config(const ov::snippets::lowered::ExpressionPtr& expr,
+                                              const ov::snippets::lowered::LinearIRCPtr& linear_ir,
+                                              BrgemmCopyBKernelConfig& config) const {
+    const auto& input_desc = expr->get_input_port_descriptor(0);
+    const auto& output_desc = expr->get_output_port_descriptor(0);
+
+    // Need to update K, N
+    // 1. If the original value in subtensor is `FULL_DIM`, it means that
+    //    BrgemmCopyB block should process full tensor by this dim -> take dimension from shape
+    // 2. Otherwise, BrgemmCopyB block processes part of the tensor by this dim
+    //    (there is blocking by this dimension) -> take from Loop increment
+
+    const auto planar_shape = ov::snippets::utils::get_planar_vdims(expr->get_input_port(0));
+    const auto& in_subtensor = input_desc->get_subtensor();
+
+    size_t loop_idx = 0;
+    const auto& loop_ids = expr->get_loop_ids();
+    const auto& loop_manager = linear_ir->get_loop_manager();
+
+    auto init = [&](size_t& dim, size_t& blk, size_t idx) {
+        dim = *(planar_shape.rbegin() + idx);
+        blk = *(in_subtensor.rbegin() + idx);
+        if (ov::snippets::utils::is_full_dim_value(blk)) {
+            blk = dim;
+        } else {
+            OPENVINO_ASSERT(loop_idx < loop_ids.size(), "Loop is missed");
+            const auto& current_expanded_loop_info = loop_manager->get_loop_info<ov::snippets::lowered::ExpandedLoopInfo>(loop_ids[loop_idx++]);
+            blk = current_expanded_loop_info->get_increment();
```

Comment:
BTW, why do we read block size from the loop increment? Can the increment change in runtime depending on shapes?

Reply:
Ok, tail processing, fair enough.